plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -817,4 +922,202 @@ private Map<String, String> getTableProperties(String schemaName, String tableNa
+        assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN col SET DATA TYPE integer");
+        assertEquals(getColumnType(tableName, "col"), "integer");
+    }
+
+    @Test
+    @Override
+    public void testSetColumnOutOfRangeType()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_SET_COLUMN_TYPE) && hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+
+        String tableName = "test_set_column_out_of_range";
+        assertUpdate("CREATE TABLE " + tableName + " (col bigint, col2 int not null) WITH (order_by=ARRAY['col2'], engine = 'MergeTree')");
+        query("insert into " + tableName + " values(9223372036854775807, 22)");
+        assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN col SET DATA TYPE integer");
+
+        assertEquals(getColumnType(tableName, "col"), "integer");
+        assertQuery("SELECT col FROM " + tableName, "VALUES -1");
+    }
+
+    @Test
+    @Override
+    public void testSetColumnTypeWithComment()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_SET_COLUMN_TYPE) && hasBehavior(SUPPORTS_CREATE_TABLE_WITH_COLUMN_COMMENT));
+        String tableName = "test_set_column_with_comment";
+
+        assertUpdate("CREATE TABLE " + tableName + " (col bigint COMMENT 'test comment', col2 int not null) WITH (order_by=ARRAY['col2'], engine = 'MergeTree')");
+        assertEquals(getColumnComment(tableName, "col"), "test comment");
+
+        assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN col SET DATA TYPE bigint");
+        assertEquals(getColumnComment(tableName, "col"), "test comment");
+    }
+
+    @Test
+    @Override
+    public void testSetColumnTypeWithDefaultColumn()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_SET_COLUMN_TYPE) && hasBehavior(SUPPORTS_INSERT));
+        try (TestTable table = createTableWithDefaultColumns()) {
+            // col_default column inserts 43 by default
+            assertUpdate("ALTER TABLE " + table.getName() + " ALTER COLUMN col_default SET DATA TYPE bigint");
+            assertUpdate("INSERT INTO " + table.getName() + " (col_required, col_required2) VALUES (1, 10)", 1);
+            assertQuery("SELECT col_default FROM " + table.getName(), "VALUES 43");
+        }
+    }
+
+    @Test
+    @Override
+    public void testSetColumnTypeWithNotNull()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_SET_COLUMN_TYPE) && hasBehavior(SUPPORTS_NOT_NULL_CONSTRAINT));
+
+        String tableName = "test_set_column_with_not_null";
+
+        assertUpdate("CREATE TABLE " + tableName + " (col bigint not null, col2 int not null) WITH (order_by=ARRAY['col2'], engine = 'MergeTree')");
+        assertFalse(columnIsNullable(tableName, "col"));
+
+        assertUpdate("ALTER TABLE " + tableName + " ALTER COLUMN col SET DATA TYPE bigint");
+        assertFalse(columnIsNullable(tableName, "col"));
+    }
+
+    @DataProvider
+    @Override
+    public Object[][] setColumnTypesDataProvider()
```

Comment:
Why do you need to override this method? Please override `filterSetColumnTypesDataProvider` instead.

Reply:
overriding `filterSetColumnTypesDataProvider` doesnt seem to work.
Because the `setColumnTypeSetupData` is called from `setColumnTypesDataProvider`

```   @DataProvider
    public Object[][] setColumnTypesDataProvider()
    {
        return setColumnTypeSetupData().stream()
                .map(this::filterSetColumnTypesDataProvider)
                .flatMap(Optional::stream)
                .collect(toDataProvider());
    }
```

Reply:
Please take a look at other connectors. e.g. `TestIcebergParquetConnectorTest`